### PR TITLE
Display module images everywhere

### PR DESCRIPTION
**Pull request by jfdoming**
## Implementation Description
Display module images everywhere

## Screenshots
Visual depiction of PR

## What should reviewers focus on?
- 

## Testing Procedure
- What test(s) should we run to make sure your code works?
   - What ways could the input be wrong?
   - Is it secure? Could nefarious users access/break it?

## Things to Note
- Additional dependencies/libraries you've added?
- Things you'd want to know when coming back to the code after a few months

## Checklist
- [ ] Ensure code follows style guide by running the linters
- [ ] I have updated the documentation or deemed it unnecessary
- [ ] I have linked the relevant issue in this PR
- [ ] I have requested a review from the PL, as well as other dev(s) (and designers if necessary)

### CODE REVIEW

File: backend/services/implementations/courseService.ts [hunk 1]
```diff
@@ -34,7 +35,48 @@ const publishedModulesOnlyQuery = [
...
 
 class CourseService implements ICourseService {
+  imageService: IImageUploadService;
+
+  constructor(imageService: IImageUploadService) {
+    this.imageService = imageService;
+  }
+
   /* eslint-disable class-methods-use-this */
+  private async hydrateImages(
```
Comment (draymond63) [inline backend/services/implementations/courseService.ts:45]: Just so I understand, what is the point of the backend hydration? Are you downloading the image in the backend so all the frontend has to do is use the attribute?

File: backend/services/implementations/courseService.ts [hunk 2]
```diff
@@ -34,7 +35,48 @@ const publishedModulesOnlyQuery = [
...
+    course: CourseResponseDTO,
+  ): Promise<CourseResponseDTO> {
+    const coursePreviewImage = course.image
+      ? await this.imageService.download(course.image)
+      : course.previewImage;
```
Comment (draymond63) [inline backend/services/implementations/courseService.ts:50]: Do courses have `image` and `previewImage` fields now? Is this new?

File: backend/services/implementations/courseService.ts [hunk 3]
```diff
@@ -34,7 +35,48 @@ const publishedModulesOnlyQuery = [
...
+    const newModules = await Promise.all(
+      course.modules.map(async (module) => {
+        const modulePreviewImage = module?.image
+          ? await this.imageService.download(module.image)
+          : module?.previewImage;
```
Comment (draymond63) [inline backend/services/implementations/courseService.ts:55]: I know when I made the data model, I had put in a module `image` and `previewImage` property. Obviously I've never made use of them, but is there a meaningful distinction between the two? It seems like previewImage is being used as a backup?

**Review by draymond63 — COMMENTED** [review]
Just some general questions for my knowledge's sake :)

File: backend/services/implementations/courseService.ts (hunk 1 above)
Comment (jfdoming) [reply backend/services/implementations/courseService.ts:45]: Kind of. Firebase uses temporary signed URLs based on a "file path" to access images, but these expire every hour so we can't store them. Each time we want to access an image, we have to recreate this URL (or use one from within the hour). So it's not downloading the image so much as getting the image URL.

**Review by draymond63 — APPROVED** [review]
Sweet! Looks good :)

**Pull request merged**